runtime/tests/decimal_operations.c

```diff
@@ -0,0 +1,59 @@
+#include <string.h>
+#include <assert.h>
+#include "../balrt.h"
+#include "../hash.h"
+#include "../third-party/decNumber/decQuad.h"
+
+HASH_DEFINE_KEY;
+
+void validate(TaggedPtr tp, const char *decStr) {
```

Comment:
I suggest you call this decimal_arith.c and use it for all arithmetic operations.

Reply:
Changed